src/test/groovy/unit/org/pih/warehouse/invoice/PrepaymentInvoiceMigrationServiceSpec.groovy

```diff
@@ -0,0 +1,272 @@
+        given:
+        Order order = new Order(
+                paymentTerm: new PaymentTerm(prepaymentPercent: 100),
+        )
+        OrderAdjustment adjustment = new OrderAdjustment(
+                canceled: true,
+                order: order,
+        )
+
+        String invoiceItemId = "1"
+        new Invoice(
+                invoiceType: prepaymentInvoiceType,
+                invoiceItems: [
+                        new InvoiceItem(
+                                id: invoiceItemId,
+                                quantity: 1,
+                                quantityPerUom: 1,
+                                orderAdjustments: [adjustment],
+                        ),
+                ],
+        ).save(validate: false)
+
+        when:
+        service.updateAmountForPrepaymentInvoiceItems()
+
+        then:
+        assert InvoiceItem.findById(invoiceItemId).amount == 0
+    }
+
+    void 'updateAmountForPrepaymentInvoiceItems correctly sets amount for an adjustment to an order item'() {
+        given:
+        Order order = new Order(
+                paymentTerm: new PaymentTerm(prepaymentPercent: prepaymentPercent),
+        )
+        OrderItem orderItem = new OrderItem(
+                order: order,
+                quantity: quantity,
+                unitPrice: unitPrice,
+        )
+        OrderAdjustment adjustment = new OrderAdjustment(
+                order: order,
+                orderItem: orderItem,
+                amount: adjustmentAmount,
+                percentage: adjustmentPercent,
+        )
+
+        String invoiceItemId = "1"
+        new Invoice(
+                invoiceType: prepaymentInvoiceType,
+                invoiceItems: [
+                        new InvoiceItem(
+                                id: invoiceItemId,
+                                quantity: 0,
+                                quantityPerUom: 0,
+                                orderItems: [orderItem],
+                                orderAdjustments: [adjustment],
+                        ),
+                ],
+        ).save(validate: false)
+
+        when:
+        service.updateAmountForPrepaymentInvoiceItems()
+
+        then:
+        assert InvoiceItem.findById(invoiceItemId).amount == expectedAmount
+
+        where:
+        quantity | unitPrice | prepaymentPercent | adjustmentPercent | adjustmentAmount || expectedAmount
+        1        | 1.0       | 100               | 100               | 1                || 1
+        2        | 1.0       | 100               | 100               | 1                || 1
+        1        | 2.0       | 100               | 100               | 1                || 1
+        1        | 1.0       | 50                | 100               | 1                || 0.5
+        1        | 1.0       | 100               | 50                | 1                || 1
+        1        | 1.0       | null              | 100               | 1                || 1
+
```

Comment:
Is there an empty line, or is it just odd GitHub formatting?

Reply:
Yeah I put empty lines to split up the logical groups (where adjustmentAmount is null vs not-null). I can probably add a comment here to explain why.